Run repair steps on uninstall in all cases

LEDfan opened the pull request:
Fixes #5832 

Maybe it's a good idea to mark `\OC_app::disable` as deprecated?

LEDfan committed:
Run repair steps on uninstall in all cases

Signed-off-by: Tobia De Koninck <tobia@ledfan.be>

LEDfan commented:
BTW it looks like the `\OC_app::enable` method actually also installs and downloads the app, maybe it's a good idea to move this code a `downloadAndInstallApp` method inside the appmanager and then call this method inside the enable method command? (https://github.com/nextcloud/server/blob/8d83e239f321ef63e08c2c1a6658449c7376c46f/core/Command/App/Enable.php#L78-L78)

And another interesting thing is that the `installApp` method of `OC_App` is not used according to PHPstorm and grep.

juliushaertl commented:
> Maybe it's a good idea to mark \OC_app::disable as deprecated?

Yes, let's get rid of that. Can you add that and check for other usages in the server code?

> BTW it looks like the \OC_app::enable method actually also installs and downloads the app, maybe it's a good idea to move this code a downloadAndInstallApp method inside the appmanager and then call this method inside the enable method command? 

Maybe that is more something for the Installer class. We should keep the install code separated from the app management.

> And another interesting thing is that the installApp method of OC_App is not used according to PHPstorm and grep.

Good catch, although it has already been removed by a68895e33427483f149e55ae3d6ec30315d8ac96 :wink:

LEDfan committed:
Remove OC_App::disable method

Signed-off-by: Tobia De Koninck <tobia@ledfan.be>

LEDfan committed:
Remove unused $enableAppsCache in OC_App

Signed-off-by: Tobia De Koninck <tobia@ledfan.be>

LEDfan commented:
> Yes, let's get rid of that. Can you add that and check for other usages in the server code?

Done, I removed the method because it's a private API.

I also removed the `$enableAppsCache` because it wasn't used anymore.

MorrisJobke commented:
> I also removed the $enableAppsCache because it wasn't used anymore.

👍 



> Yes, let's get rid of that. Can you add that and check for other usages in the server code?

Looks good and I also didn't found any other apps using it.

MorrisJobke commented:
Let me rebase and review this one here.

MorrisJobke reviewed (approved):
Tested and works 👍

MorrisJobke commented:
@nickvergessen @rullzer Please review as this is some crucial cleanup.

MorrisJobke commented on lib/private/App/AppManager.php:
```diff
@@ -287,6 +287,16 @@ public function disableApp($appId) {
+		}
+
+		// emit disable hook - needed anymore ?
+		\OC_Hook::emit('OC_App', 'pre_disable', array('app' => $appId));
```
I'm also fine with removing this, because I couldn't find any reference to this in any app or in the server itself. Also there is a new proper event dispatched just two lines below.

nickvergessen reviewed (approved):
Don't see any issues and +1 for removing the trap (calling appmanager directly skips uninstall repair steps)

MorrisJobke committed:
Remove unused and outdated hook

Signed-off-by: Morris Jobke <hey@morrisjobke.de>
